text/0000-ember-data-identifiers.md

```diff
@@ -0,0 +1,194 @@
+- Start Date: 2018-11-25
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Ember Data Record Identifiers 
+
+## Summary
+
+A follow-up RFC to [RecordData]() that simplifies APIs and ensures capabilities that were intended by the original RFC but not delivered.
+
+Specifically it ensures that `RecordData` can be implemented as a `singleton`, eliminates several redundant APIs that creeped into the original implementation, and simplifies the method signatures of `RecordData` APIs.
+
+The primary shift is that we would now utilize `json-api` resource identifiers and upgrade them within the store to `RecordIdentifier`s to communicate `type` `id` and `lid` information between APIs. We also deprecate use of `clientId` in favor of `lid`.
+
+## Motivation
+
+- nested saves / transactions
+- websockets
+- reload
+- better cache serialization
+- simpler cache and relationship management
+
+## Detailed design
+
+### New Typings
+
+```typescript
+import { Value as JSONValue } from 'json-typescript';
+
+export type Meta = { [k: string]: JSONValue };
+
+export interface ResourceIdentifier {
+  lid?: string;
+  id: string | null;
+  type: string;
+  meta?: Meta;
+}
+
+export interface RecordIdentifier {
+    lid: string;
+    id: string | null;
+    type: string;
+    meta: Meta | null;
+}
+```
+
+### New APIs.
+
+* **recordIdentifierFor**
+
+`import { recordIdentifierFor } from '@ember-data/store';`
+
+```typescript
+/*
+  Returns the RecordIdentifier for the given ResourceIdentifier.
+
+  Specifically this means that we:
+
+  - validate the `id` `type` and `lid` combo against known identifiers
+  - return an object that is guaranteed to have the `lid` for the given `type` and `id`
+  - return an object that also contains the stored `meta`
+  - this object is referentially stable
+*/
+export function recordIdentifierFor(store: Store, resourceIdentifier: ResourceIdentifier): RecordIdentifier {}
+```
+
+### Changes to existing APIs.
+
+**Store Methods**
+* `recordDataFor(modelName, id, clientId)`
+  * New Re-Arg: `recordDataFor(identifier)`
```

Comment:
Typing the identifier here would help differentiate between the `resourceIdentifier` and `recordIdentifier`

Reply:
I will once we have resolved which it should be, this question is in `unresolved questions` atm.